### ISSUE TITLE
Why tekton?

### ISSUE THREAD

**Issue by emichaf** [tektoncd/pipeline#907]
I wonder what's the difference between tekton and argo cd?

**Comment by Gl4di4torRr** [comment]
@emichaf here are my two cents...

tekton was donated to the (cd foundation)[https://cd.foundation/] which is a collaborative effort across  organizations (vendor neutral) to help improve the CI/CD space. Argo is not a member of the cd foundation. Because of tekton's donation to the cd foundation, I have been looking at tekton over something like argo.

**Comment by bobcatfish** [comment]
It's a good question @emichaf and @Gl4di4torRr has a good point :D

You might find https://github.com/tektoncd/pipeline/issues/644 interesting as well, where @RXminuS asked a very similar question (the original title was "Why not contribute to Argo?")

> I wonder what's the difference between tekton and argo cd?

Specifically I would say that argo cd is an example of the sort of tool that could make use of Tekton - we chatted with some of the Argo CD folks at this most recent Kubecon and it sounds like there might be some potential for Argo CD to use some of the things we're working on in Tekton :crossed_fingers: 

+ @alexmt @dthomson25

So long story short hopefully it won't be a Tekton OR <CI/CD tool> question forever, ideally you'll eventually be able to use Tekton AND <CI/CD tool> :)

**Comment by alexmt** [comment]
I agree with you @bobcatfish. Argo CD does not provide an ability to define a pipeline. Instead, it is supposed to be integrated with an existing pipeline tool like Tekton: https://argoproj.github.io/argo-cd/user-guide/ci_automation/

Going through tekton tutorial and trying to run my first `hello world` task today :) . Next step is to submit PR with gitops/argocd specific shared tasks to https://github.com/tektoncd/catalog

**Comment by bobcatfish** [comment]
I'm gonna close this PR for now but feel free to re-open and/or open more issues if we should discuss further. Thanks @emichaf !

**Comment by emichaf** [comment]
Hi again,
Wirh Argo ci you can specify workflows with steps, also different gateways exist (rest, amqp..) to trigger workflows, build steps are executed in containers.possible to handle input/output in steps with different artifact etc like git...

Is a tekton use case to re-use loosely coupled steps in different pipelines?  

Hard to see the differences between Argo and Tekton!

What will Tekton solve which are not included in Argo CI & CD ?

**Comment by alexmt** [comment]
@AdityaGupta1 just sent a PR with the first argocd task: https://github.com/tektoncd/catalog/pull/49

**Comment by AliArdaOrhan** [comment]
This question is still unanswered and waits for someone to create a value proposition for Tekton.
